validations.js

```diff
@@ -1,82 +1,44 @@
- * @param {*} tpl - astObject
- * @param {*} value - value assigned to parseAssign statement
- */
-function validateJSON(engine, tpl, value, validationErrors) {
-  try {
-    const trimmedValue = value.trim();
-
-    if (!Lexical.isLiteral(trimmedValue)) {
-      throw new Error("Invalid value assigned to parseAssign statement");
-    }
-    const evalValue = engine.evalValue(trimmedValue, {});
-    const jsonStr = `{"val": ${
-      typeof evalValue === "string" ? JSON.stringify(evalValue) : evalValue
-    }}`;
-
-    JSON.parse(jsonStr);
-  } catch (e) {
-    validationErrors.push({
-      expression: `${tpl.name} ${tpl.tagImpl.key} = ${tpl.tagImpl.value}`,
-      errorMessage: `${e.message} at line ${tpl.token.line}`,
-    });
-  }
-}
-
-function parseIf(ifTemplate, callback) {
-  const impl = ifTemplate.tagImpl;
-  impl.branches.forEach(function (branch) {
-    callback(branch.templates);
-  });
-  if (impl.elseTemplates) {
-    callback(impl.elseTemplates);
-  }
-}
-
-function parseForOrUnless(forTemplate, callback) {
-  const impl = forTemplate.tagImpl;
-  if (impl.templates) {
-    callback(impl.templates);
-  }
-  if (impl.elseTemplates) {
-    callback(impl.elseTemplates);
-  }
-}
-
-/**
- - checks if valid JSON object has been passed to parseAssign statement
+ - checks if valid dynamic table by looking for presence of atleast one $$answer = something which is not present inside any loop or conditional blocks
  - adds errors to validation errors array
  * @param {*} expression - liquid expression
  */
-function checkValidJSON(engine, expression) {
+function checkValidDynamicTable(engine, expression) {
+  const validationErrors = [];
   const templates = getTemplates(engine, expression);
-  let validationErrors = [];
 
-  function processTemplates(templates) {
-    templates.forEach((tpl) => {
-      if (tpl.name === "parseAssign") {
-        validateJSON(engine, tpl, tpl.tagImpl.value, validationErrors);
-      } else if (tpl.name === "if") {
-        parseIf(tpl, processTemplates);
-      } else if (tpl.name === "for" || tpl.name === "unless") {
-        parseForOrUnless(tpl, processTemplates);
-      }
-    });
-  }
+  const dynamicTableTags = templates.filter(
+    (tpl) => tpl.type === "tag" && tpl.name === "computeColumn"
+  );
+
+  dynamicTableTags.forEach((table) => {
+    if (!checkIfTargetStatementExists(table.tagImpl?.templates)) {
+      validationErrors.push({
+        message:
+          "$$answer is not assigned outside any loops or condition block",
+        metadata: {
+          expression,
+          tableName: table.tagImpl?.tableName,
+          columnName: table.tagImpl?.columnName,
+        },
+      });
+    }
+  });
 
-  processTemplates(templates);
   return validationErrors;
 }
 
+function checkIfTargetStatementExists(templates) {
+  return templates.some((tpl) => {
+    return tpl.type === "tag" && tpl.tagImpl?.key?.trim() === "$$answer";
+  });
+}
+
 module.exports = {
-  checkValidJSON,
```

Comment:
Why did you remove this?

Reply:
Let's create individual files for each validation
- `validations/paseAssign.js`
- `validations/computeColumn.js`

And then export all of these from `validations.js`